cpp/src/wholegraph_ops/raft_random_gen.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2019-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <wholememory/wholegraph_op.h>
+#include <cmath>
+#include <wholememory_ops/raft_random.cuh>
+
+
+#include "error.hpp"
+#include "logger.hpp"
+
+// generate_negative=true: generate uint32 data 
+wholememory_error_code_t generate_random_positive_int_cpu(
+  int64_t random_seed,
+  int64_t subsequence,
+  wholememory_tensor_t output
+) {
+  auto output_tensor_desc = *wholememory_tensor_get_tensor_description(output);
+  if (output_tensor_desc.dim != 1) {
+    WHOLEMEMORY_ERROR("output should be 1D tensor.");
+    return WHOLEMEMORY_INVALID_INPUT;
+  }
+  if (output_tensor_desc.dtype != WHOLEMEMORY_DT_INT64 && output_tensor_desc.dtype != WHOLEMEMORY_DT_INT) {
+    WHOLEMEMORY_ERROR("output should be int64 or int32 tensor.");
+    return WHOLEMEMORY_INVALID_INPUT;
+  }
+
+  auto* output_ptr = wholememory_tensor_get_data_pointer(output);
+  PCGenerator rng((unsigned long long)random_seed, subsequence, 0);
+  for (int64_t i = 0; i < output_tensor_desc.sizes[0]; i++) { 
+    if (output_tensor_desc.dtype == WHOLEMEMORY_DT_INT) {
+      int random_num;
```

Comment:
for consistency and safety w.r.t. C++ implementations, it may make sense to use `int32_t` here. This isn't important though